src/plugin/task-runner.go

```
@@ -57,8 +58,9 @@ func (trp TaskRunnerPlugin) Run(ctx context.Context, fetcher ConfigFetcher) erro
 		// TODO: This is currently a magic number. If we want this to be configurable, remove the nolint directive and fix it up
 		o.MaxDelay = 10 * time.Second //nolint:mnd
 	})
-	result, err := awsinternal.WaitForCompletion(ctx, waiterClient, taskArn)
+	result, err := awsinternal.WaitForCompletion(ctx, waiterClient, taskArn, config.TimeOut)
 	if err != nil {
+		_ = buildKiteAgent.Annotate(ctx, fmt.Sprintf("Task did not complete successfully within timeout %v", result.Failures[0]), "error", "ecs-task-runner")
```

I'm discarding this error because we're already erroring here. It's a bit late in the afternoon and I don't know how to handle this in an idiomatic way with my frazzled brain.

---

Let's leave it for tomorrow/next week in that case.

---

Looking at this: the two messages may not be accurate together.

Some ideas/thoughts:
- Do we know if a timeout is considered a failure/error?
  - If it's a failure, can we check the Failure contents and return a `fmt.Errorf` using the newly-added message, or the generic.
  - If timeouts are not a failure, would a concurrent waiter function be suitable here that returns a `fmt.Error`? (I don't personally like this idea at all, besides adding complexity, it just sounds like dookie design)
  - This could all be answered with: What happens when a timeout occurs in this `WaitForCompletion` function?
- Change up the messaging entirely (coward's way out 😤)

---

>Looking at this: the two messages may not be accurate together.
>
>Some ideas/thoughts:
>
>Do we know if a timeout is considered a failure/error?
>If it's a failure, can we check the Failure contents and return a fmt.Errorf using the newly->added message, or the generic.
>If timeouts are not a failure, would a concurrent waiter function be suitable here that >returns a fmt.Error? (I don't personally like this idea at all, besides adding complexity, it >just sounds like dookie design)
>This could all be answered with: What happens when a timeout occurs in this WaitForCompletion >function?
>Change up the messaging entirely (coward's way out 😤)

Yeah, so a few things that ultimately change how we need to approach this:

- Timeouts do not return any actual ECS error messages. Since the type that interacts with them is actually divorced from the infra side of things we only get error details in scenario that something goes wrong upstream - a task is unable to start due to capacity provider config, for instance.
- Instead we get an error returned: `fmt.Errorf("exceeded max wait time for TasksStopped waiter")` (ref: `go/pkg/mod/github.com/aws/aws-sdk-go-v2/service/ecs@v1.49.0/api_op_DescribeTasks.go:569`). 
- In fact, you cannot expect a `*DescribeTasksOutput` in any circumstance where the wait method returns an error. So, we should assume that errors are actually going to be issues encountered in the method itself, rather than anything upstream. The inverse is only partially true, we are going to have cases where the task succeeds end-to-end, and others where a `DescribeTasksOutput.Failure` array is populated.
- I'm thinking we can use this as a sentinel error to annotate the build in this specific case, and we should annotate with any failure messages that are returned in the case where the return is successful.